src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -978,6 +980,90 @@ void goto_instrument_parse_optionst::instrument_goto_program()
   if(cmdline.isset("remove-function-pointers"))
     do_function_pointer_removal();
 
+  if(cmdline.isset("function-inline"))
+  {
+    std::string function=cmdline.get_value("function-inline");
+    assert(!function.empty());
+
+    do_function_pointer_removal();
+
+    status() << "Inlining calls of function `" << function << "'" << eom;
+
+    if(!cmdline.isset("log"))
+    {
+      goto_function_inline(
+        goto_functions,
+        function,
+        ns,
+        ui_message_handler,
+        true);
+    }
+    else
+    {
+      std::string filename=cmdline.get_value("log");
+      bool have_file=!filename.empty() && filename!="-";
+
+      jsont result=
+        goto_function_inline_and_log(
+          goto_functions,
+          function,
+          ns,
+          ui_message_handler,
+          true);
+
+      if(have_file)
+      {
+#ifdef _MSC_VER
+        std::ofstream of(widen(filename));
+#else
+        std::ofstream of(filename);
+#endif
+        if(!of)
+          throw "failed to open file "+filename;
+
+        of << result;
+        of.close();
+      }
+      else
+      {
+        std::cout << result << std::endl;
+      }
+    }
+
+    goto_functions.update();
+    goto_functions.compute_loop_numbers();
+  }
+
+  if(cmdline.isset("partial-inline"))
+  {
+    do_function_pointer_removal();
+
+    status() << "Partial inlining" << eom;
+    goto_partial_inline(goto_functions, ns, ui_message_handler, true);
+
+    goto_functions.update();
+    goto_functions.compute_loop_numbers();
+  }
+
+  // now do full inlining, if requested
+  if(cmdline.isset("inline"))
+  {
+    status() << "Function Pointer Removal" << eom;
+    remove_function_pointers(
+      symbol_table, goto_functions, cmdline.isset("pointer-check"));
+
+    if(cmdline.isset("show-custom-bitvector-analysis") ||
+       cmdline.isset("custom-bitvector-analysis"))
+    {
+      do_remove_returns();
+      thread_exit_instrumentation(goto_functions);
+      mutex_init_instrumentation(symbol_table, goto_functions);
+    }
+
+    status() << "Performing full inlining" << eom;
+    goto_inline(goto_functions, ns, ui_message_handler, true);
+  }
```

Comment:
I am surprised this code is marked as new, I thought we had had it before already?